test/Microsoft.AspNet.Razor.Runtime.Test/DefaultTagHelperContentTest.cs

```diff
@@ -0,0 +1,201 @@
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+
+            // Act
+            tagHelperContent.SetContent(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.ToString());
+        }
+
+        [Fact]
+        public void SetContentClearsExistingContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+
+            // Act
+            tagHelperContent.SetContent("Contoso");
+            tagHelperContent.SetContent(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.ToString());
+        }
+
+        [Fact]
+        public void CanGetContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+            tagHelperContent.SetContent(expected);
+
+            // Act
+            var actual = tagHelperContent.GetContent();
+
+            // Assert
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void CanAppendContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+
+            // Act
+            tagHelperContent.Append(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.GetContent());
+        }
+
+        [Fact]
+        public void CanIdentifyIfModified()
+        {
+            // Arrange
+            var tagHelperContent1 = new DefaultTagHelperContent();
+            var tagHelperContent2 = new DefaultTagHelperContent();
+            var tagHelperContent3 = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent1.SetContent("Test");
+            tagHelperContent2.Append("Foo");
+            tagHelperContent3.Clear();
+
+            // Assert
+            Assert.True(tagHelperContent1.IsModified);
+            Assert.True(tagHelperContent2.IsModified);
+            Assert.True(tagHelperContent3.IsModified);
+        }
+
+        [Fact]
+        public void CanAddWhiteSpaces()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("  ");
+
+            // Assert
+            Assert.Equal(2, tagHelperContent.ToString().Length);
+        }
+
+        [Fact]
+        public void CanIdentifyWhiteSpaces()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("  ");
+            tagHelperContent.Append(" ");
+
+            // Assert
+            Assert.True(tagHelperContent.IsWhiteSpace);
+        }
+
+        [Fact]
+        public void CanIdentifyIsEmpty()
```

Comment:
Should be able to append `string.Empty` multiple times and still have `IsEmpty` be `true`, should add a test.